lifinance/sdk
# fix: improvements in token balance fetching methods

# FEATURE

Improve SDK getTokenBalancesForChains and getTokenBalances methods [LF-3225]

## TL;DR

Add a boolean parameter to the token balance methods to allow users to hide zero balance tokens. Also, improve error handling for provider errors during token balance fetching.

## DESCRIPTION

This feature improves the SDK's `getTokenBalancesForChains` and `getTokenBalances` methods by adding a boolean parameter. This parameter allows users to specify whether they want to hide zero balance tokens or not. By default, all tokens are returned, but users can choose to exclude tokens with zero balances by setting the boolean parameter to `true`.

Additionally, this feature enhances the error handling for provider errors during token balance fetching. It throws the correct validation error when encountering issues with the provider during the retrieval of token balances.

## STEPS TO TEST

1. Call the `getTokenBalancesForChains` and `getTokenBalances` methods with the boolean parameter set to `true` to hide zero balance tokens.
2. Verify that the returned token balances exclude tokens with zero balances.

## EXPECTED RESULTS

- When setting the boolean parameter to `true`, the token balance methods should return balances excluding zero balance tokens.
- When encountering provider errors during token balance fetching, the methods should throw the correct validation error.

## CHECKLIST

- [ ] I have commented the code I wrote to ensure clarity.
- [ ] I have added tests that cover the new functionality.
- [ ] I have made sure that existing tests are still passing.


[LF-3225]: https://lifi.atlassian.net/browse/LF-3225?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
